### PR TITLE
Bump elixir to 1.16.0 and Erlang OTP to 26.2.1

### PR DESCRIPTION
**Pull request by k1rill-fedoseev**
## Changelog

* Migrate to the latest elixir stack versions:
  * Elixir 1.16.0
  * Erlang OTP 26.2.1
* Fix various compilation/linter/credo/dialyzer warnings
* Fix compilation issues coming from `prometheus_process_collector` dependency
  * Compilation on Apple Silicon chips (https://github.com/deadtrickster/prometheus_process_collector/pull/24)
  * Compilation on OTP 26 (https://github.com/deadtrickster/prometheus_process_collector/pull/30)

## Checklist for your Pull Request (PR)

  - [ ] I added an entry to `CHANGELOG.md` with this PR
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [ ] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [ ] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [ ] If I add new indices into DB, I checked, that they are not redundant with PGHero or other tools

### CODE REVIEW

File: apps/indexer/test/indexer/buffered_task_test.exs [hunk 1]
```diff
@@ -243,7 +243,8 @@ defmodule Indexer.BufferedTaskTest do
                  flush_timer: nil,
                  task_supervisor: BufferedTaskSup,
                  max_batch_size: 1,
-                 max_concurrency: 1
+                 max_concurrency: 1,
+                 poll: false
```
Comment (k1rill-fedoseev) [inline apps/indexer/test/indexer/buffered_task_test.exs:247]: Seems like the test didn't properly work in elixir 1.14, didn't manage to debug the reason why.
Without `poll: false`, BufferedTask will attempt to call init/2 when given an empty queue.

File: apps/explorer/test/explorer/exchange_rates/source/coin_gecko_test.exs [hunk 2]
```diff
@@ -143,7 +143,7 @@ defmodule Explorer.ExchangeRates.Source.CoinGeckoTest do
       Application.put_env(:explorer, CoinGecko, base_url: "http://localhost:#{bypass.port}")
 
       on_exit(fn ->
-        Application.put_env(:explorer, :coin, "POA")
+        Application.put_env(:explorer, :coin, "ETH")
```
Comment (k1rill-fedoseev) [inline apps/explorer/test/explorer/exchange_rates/source/coin_gecko_test.exs:146]: Setting env to a non-default value breaks test sometimes in a different file.

File: apps/explorer/lib/explorer/account/notifier/email.ex [hunk 3]
```diff
@@ -118,15 +117,15 @@ defmodule Explorer.Account.Notifier.Email do
   end
 
   defp address_url(address_hash) do
-    Helpers.address_url(uri(), :show, address_hash)
+    uri() |> URI.append_path("/address/#{address_hash}") |> to_string()
```
Comment (k1rill-fedoseev) [inline apps/explorer/lib/explorer/account/notifier/email.ex:120]: Elixir doesn't like cyclic dependencies now, so this is probably the easiest way to remove Explorer -> BlockScoutWeb dependency

File: apps/block_scout_web/mix.exs [hunk 4]
```diff
@@ -114,8 +114,11 @@ defmodule BlockScoutWeb.Mixfile do
       {:prometheus_phoenix, "~> 1.2"},
       # Expose metrics from URL Prometheus server can scrape
       {:prometheus_plugs, "~> 1.1"},
-      # OS process metrics for Prometheus
-      {:prometheus_process_collector, "~> 1.3"},
+      # OS process metrics for Prometheus, custom ref to include https://github.com/deadtrickster/prometheus_process_collector/pull/30
+      {:prometheus_process_collector,
```
Comment (k1rill-fedoseev) [inline apps/block_scout_web/mix.exs:118]: There is a new upstream release for prometheus_ex, btw - https://github.com/deadtrickster/prometheus.ex/releases/tag/v3.1.0
We might wanna update in a separate PR.

**Comment by k1rill-fedoseev** [comment]
One more randomly-failing test popped up in `test/explorer/market/market_test.exs:20`, can't reproduce it locally...

File: apps/explorer/test/explorer/market/market_test.exs [hunk 5]
```diff
@@ -18,6 +18,8 @@ defmodule Explorer.MarketTest do
   end
 
   test "fetch_recent_history/1" do
+    ConCache.delete(:market_history, :last_update)
```
Comment (k1rill-fedoseev) [inline apps/explorer/test/explorer/market/market_test.exs:21]: Another test cached an empty value causing this test to fail.

**Review by vbaranov — COMMENTED** [review]
@k1rill-fedoseev could you please rebase to master branch? There are merging conflicts currently: I recently moved OTP_VERSION and ELIXIR_VERSION to github envs and re-used them in Github actions workflows.

**Review by sl1depengwyn — APPROVED** [review]
